pilot/pkg/security/model/authentication.go

```diff
@@ -128,17 +128,34 @@ func AppendURIPrefixToTrustDomain(trustDomainAliases []string) []string {
 // ApplyToCommonTLSContext completes the commonTlsContext
 func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, proxy *model.Proxy,
 	subjectAltNames []string, crl string, trustDomainAliases []string, validateClient bool,
+	tlsCertificates []*networking.ServerTLSSettings_TLSCertificate,
 ) {
+	sdsSecretConfigs := make([]*tls.SdsSecretConfig, 0)
 	customFileSDSServer := proxy.Metadata.Raw[security.CredentialFileMetaDataName] == "true"
+	// Envoy does not support client validation using multiple CA certificates.
+	// So we only use the first one.
```

Comment:
nit: fix the comment. We do not use the first one. We use what is specified in ServerSettings